app/controllers/presence_controller.rb

```diff
@@ -5,21 +5,26 @@ class PresenceController < ApplicationController
   before_action :ensure_logged_in, only: [:update]
 
   def get
-    name = params.require(:channel)
+    names = params.require(:channels)
+    raise Discourse::InvalidParameters.new(:channels) if !(names.is_a?(Array) && names.all? { |n| n.is_a? String })
 
-    begin
+    names.uniq!
+
+    raise Discourse::InvalidParameters.new("Too many channels") if names.length > 50
```

Comment:
I think we usually try to avoid magic numbers and preferably use a constant with a name that reflects what the number represents.